src/Button.js

```diff
@@ -2,38 +2,53 @@ import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import {COMMON, get} from './constants'
 import theme from './theme'
-import getButtonStyles from './ButtonStyles'
-import {layout} from 'styled-system'
+import buttonStyles from './ButtonStyles'
+import {compose, system, layout} from 'styled-system'
 import systemPropTypes from '@styled-system/prop-types'
 
-function fontSize({size = '14px', ...props}) {
-  return {
-    fontSize:
-      size === 'sm' ? `${get('fontSizes.0')(props)}px` : size === 'large' ? `${get('fontSizes.2')(props)}px` : size
-  }
+const fontSizeVariants = {
+  small: get('fontSizes.0'),
+  medium: get('fontSizes.1'),
+  large: get('fontSizes.2')
 }
 
-const Button = styled.button.attrs(props => ({
-  onClick: props.disabled ? undefined : props.onClick,
-  className: props.disabled ? 'disabled' : ''
+const variants = ({variant, theme}) => {
+  const getter = fontSizeVariants[variant] || fontSizeVariants['medium']
+  return {fontSize: getter(theme)}
```

Comment:
Why aren't we using the standard variant API here? It is a little hard to read because it seems like we are using the prop name variant but sort of re-implementing the variant function from styled-system. Could we use the standard variant API and then set defaultProps on variant to medium instead?

Reply:
I ended up here in order to keep the button rendering properly if a value other than `small`, `medium` or `large` was passed for the `variant` prop (e.g. it defaults to the `medium` value). I agree that it's more confusing, and I'm not convinced any upside is worth it.